### PR TITLE
Add Egress resource reference for tagging metrics

### PR DESCRIPTION
**Pull request by pierDipi**
To be able to tag trigger, subscription and KafkaSource metrics
we need to have the resource associated with an Egress.

https://github.com/knative-sandbox/eventing-kafka-broker/pull/1740 is based on this.

Related #1701 

Signed-off-by: Pierangelo Di Pilato <pierdipi@redhat.com>

### CODE REVIEW

File: control-plane/pkg/reconciler/channel/channel.go [hunk 1]
```diff
@@ -562,6 +576,21 @@ func (r *Reconciler) getChannelContractResource(ctx context.Context, topic strin
...
+
+	for _, s := range subscriptions {
+		if s.UID == subscriber.UID {
+			return s.Name, nil
+		}
+	}
+
+	return "", errors.New(string(metav1.StatusReasonNotFound))
+}
+
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/channel/channel.go:593]: @aliok do you have a better/faster way of doing this? Do we have a label to list subscriptions for the given channel only?

Comment (pierDipi) [reply control-plane/pkg/reconciler/channel/channel.go:600]: I checked and we don't have a label for this.

**Comment by pierDipi** [comment]
@matzew @aliok @devguyio can anyone of you please review this one?

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Pull request merged**